# Drug Consumption EDA

In [1]:
import pandas as pd
import numpy as np
import seaborn as sns

/Users/kibbles/anaconda3/envs/metis/lib/python3.7/importlib/_bootstrap.py:219: RuntimeWarning: numpy.ufunc size changed, may indicate binary incompatibility. Expected 192 from C header, got 216 from PyObject
  return f(*args, **kwds)


## Read data

In [10]:
df = pd.read_csv('drug_consumption.data')
df

age   gender  education  country  ethnicity   nscore   escore  \
1     0.49788  0.48246   -0.05921  0.96082    0.12600  0.31287 -0.57545   
2    -0.07854 -0.48246    1.98437  0.96082   -0.31685 -0.67825  1.93886   
3     0.49788 -0.48246   -0.05921  0.96082   -0.31685 -0.46725  0.80523   
4    -0.95197  0.48246    1.16365  0.96082   -0.31685 -0.14882 -0.80615   
5     0.49788  0.48246    1.98437  0.96082   -0.31685  0.73545 -1.63340   
...       ...      ...        ...      ...        ...      ...      ...   
1884 -0.95197  0.48246   -0.61113 -0.57009   -0.31685 -1.19430  1.74091   
1885 -0.95197 -0.48246   -0.61113 -0.57009   -0.31685 -0.24649  1.74091   
1886 -0.07854  0.48246    0.45468 -0.57009   -0.31685  1.13281 -1.37639   
1887 -0.95197  0.48246   -0.61113 -0.57009   -0.31685  0.91093 -1.92173   
1888 -0.95197 -0.48246   -0.61113  0.21128   -0.31685 -0.46725  2.12700   

       oscore   ascore   cscore  ...  ecstasy  heroin ketamine legalh  lsd  \
1    -0.58331 -0.91699 -0.00665  ...      CL0     CL0      CL0    CL0  CL0   
2     1.43533  0.76096 -0.14277  ...      CL4     CL0      CL2    CL0  CL2   
3    -0.84732 -1.62090 -1.01450  ...      CL0     CL0      CL0    CL0  CL0   
4    -0.01928  0.59042  0.58489  ...      CL0     CL0      CL2    CL0  CL0   
5    -0.45174 -0.30172  1.30612  ...      CL1     CL0      CL0    CL1  CL0   
...       ...      ...      ...  ...      ...     ...      ...    ...  ...   
1884  1.88511  0.76096 -1.13788  ...      CL0     CL0      CL0    CL3  CL3   
1885  0.58331  0.76096 -1.51840  ...      CL2     CL0      CL0    CL3  CL5   
1886 -1.27553 -1.77200 -1.38502  ...      CL4     CL0      CL2    CL0  CL2   
1887  0.29338 -1.62090 -2.57309  ...      CL3     CL0      CL0    CL3  CL3   
1888  1.65653  1.11406  0.41594  ...      CL3     CL0      CL0    CL3  CL3   

     meth mushrooms nicotine semer  vsa  
1     CL0       CL0      CL2   CL0  CL0  
2     CL3       CL0      CL4   CL0  CL0  
3     CL0       CL1      CL0   CL0  CL0  
4     CL0       CL0      CL2   CL0  CL0  
5     CL0       CL2      CL2   CL0  CL0  
...   ...       ...      ...   ...  ...  
1884  CL0       CL0      CL0   CL0  CL5  
1885  CL4       CL4      CL5   CL0  CL0  
1886  CL0       CL2      CL6   CL0  CL0  
1887  CL0       CL3      CL4   CL0  CL0  
1888  CL0       CL3      CL6   CL0  CL2  

[1885 rows x 31 columns]

## Bin ages:

    Value Meaning Cases Fraction
    -0.95197 18-24 643 34.11%
    -0.07854 25-34 481 25.52%
    0.49788 35-44 356 18.89%
    1.09449 45-54 294 15.60%
    1.82213 55-64 93 4.93%
    2.59171 65+ 18 0.95%
    Descriptive statistics
    Min Max Mean Std.dev.
    -0.95197 2.59171 0.03461 0.87813